From 2f986878719154adcb5d845f17d800d17e4e0009 Mon Sep 17 00:00:00 2001
From: Ranbir Singh <Ranbir.Singh3@Dell.com>
Date: Tue, 24 Jan 2023 15:02:37 +0530
Subject: [PATCH 1/1] MdeModulePkg/Universal/CapsuleRuntimeDxe: Fix DEADCODE
 Coverity issue

The function PrepareContextForCapsulePei sets PhysicalAddressBits to
32 and hence the check "if (PhysicalAddressBits <= 39 )" will always
succeed thereby making else part never coming into execution path.

Hence, remove the redundant check and deadcode.

Cc: Zhichao Gao <zhichao.gao@intel.com>
Cc: Ray Ni <ray.ni@intel.com>
REF: https://bugzilla.tianocore.org/show_bug.cgi?id=4297
Signed-off-by: Ranbir Singh <Ranbir.Singh3@Dell.com>
---
 .../CapsuleRuntimeDxe/X64/SaveLongModeContext.c          | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/MdeModulePkg/Universal/CapsuleRuntimeDxe/X64/SaveLongModeContext.c b/MdeModulePkg/Universal/CapsuleRuntimeDxe/X64/SaveLongModeContext.c
index dab297dd0a6c..9be9295594a3 100644
--- a/MdeModulePkg/Universal/CapsuleRuntimeDxe/X64/SaveLongModeContext.c
+++ b/MdeModulePkg/Universal/CapsuleRuntimeDxe/X64/SaveLongModeContext.c
@@ -141,13 +141,8 @@ PrepareContextForCapsulePei (
   //
   // Calculate the table entries needed.
   //
-  if (PhysicalAddressBits <= 39 ) {
-    NumberOfPml4EntriesNeeded = 1;
-    NumberOfPdpEntriesNeeded  = (UINT32)LShiftU64 (1, (PhysicalAddressBits - 30));
-  } else {
-    NumberOfPml4EntriesNeeded = (UINT32)LShiftU64 (1, (PhysicalAddressBits - 39));
-    NumberOfPdpEntriesNeeded  = 512;
-  }
+  NumberOfPml4EntriesNeeded = 1;
+  NumberOfPdpEntriesNeeded  = (UINT32)LShiftU64 (1, (PhysicalAddressBits - 30));
 
   if (!Page1GSupport) {
     TotalPagesNum = (NumberOfPdpEntriesNeeded + 1) * NumberOfPml4EntriesNeeded + 1;
-- 
2.36.1.windows.1

